homeassistant/components/sensor/nmbs.py

```diff
@@ -0,0 +1,207 @@
+        return 0
+
+    delta = dt_util.utc_from_timestamp(int(departure_time)) - dt_util.now()
+    return round((delta.total_seconds() / 60))
+
+
+def convert_ms_to_sec(delay=0):
+    """Calculate the delay in minutes. Delays are expressed in seconds."""
+    return round((int(delay) / 60))
+
+
+def get_ride_duration(departure_time, arrival_time, delay=0):
+    """Calculate the total travel time in minutes."""
+    duration = dt_util.utc_from_timestamp(
+        int(arrival_time)) - dt_util.utc_from_timestamp(int(departure_time))
+    duration_time = int(round((duration.total_seconds() / 60)))
+    return duration_time + convert_ms_to_sec(delay)
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up the NMBS sensor with iRail API."""
+    from pyrail import iRail
+    api_client = iRail()
+
+    name = config.get(CONF_NAME)
+    station_from = config.get(CONF_STATION_FROM)
+    station_to = config.get(CONF_STATION_TO)
+    station_live = config.get(CONF_STATION_LIVE)
+
+    sensors = [NMBSSensor(name, station_from, station_to, api_client)]
+
+    if station_live is not None:
+        sensors.append(NMBSLiveBoard(station_live, api_client))
+
+    add_entities(sensors, True)
+
+
+class NMBSLiveBoard(Entity):
+    """Get the next train from a station's liveboard."""
+
+    def __init__(self, live_station, api_client):
+        """Initialize the sensor for getting liveboard data."""
+        self._station = live_station
+        self._api_client = api_client
+        self._state = None
+
+    @property
+    def name(self):
+        """Return the sensor default name."""
+        return DEFAULT_NAME_LIVE
+
+    @property
+    def icon(self):
+        """Return the default icon or an alert icon if delays."""
+        if self._attrs is not None and int(self._attrs['delay']) > 0:
+            return DEFAULT_ICON_ALERT
+
+        return DEFAULT_ICON
+
+    @property
+    def state(self):
+        """Return sensor state."""
+        return self._state
+
+    @property
+    def device_state_attributes(self):
+        """Return the sensor attributes if data is available."""
+        if self._state is None or self._attrs is None:
+            return None
+
+        delay = convert_ms_to_sec(self._attrs["delay"])
+        departure = get_time_until(self._attrs['time'])
+
+        return {
+            'Delay': "{} minutes".format(delay) if delay > 0 else None,
+            "Vehicle ID": self._attrs['vehicle'],
+            'Occupancy': self._attrs['occupancy']['name'],
+            "Extra train": True if int(self._attrs['isExtra']) > 0 else False,
+            'Departure': "In {} minutes".format(departure),
+            ATTR_ATTRIBUTION: "https://api.irail.be/",
+        }
+
+    def update(self):
+        """Set the state equal to the next departure."""
+        liveboard = self._api_client.get_liveboard(self._station)
+        next_departure = liveboard['departures']['departure'][0]
+
+        self._attrs = next_departure
```

Comment:
Initialize `self._attrs` in init method.

Reply:
But next departure is only available in the update method? Setting attrs in init would not refresh the data when the next api call is being made?

Reply:
Set it to empty dict in init. All instance attributes should be initialized in init.